source/game_sa/CCoverPoint.cpp

```diff
@@ -1 +1,45 @@
-#include "StdInc.h"
\ No newline at end of file
+#include "StdInc.h"
+#include "CCoverPoint.h"
+
+void CCoverPoint::InjectHooks() {
+    ReversibleHooks::Install("CCoverPoint", "CanAccomodateAnotherPed", 0x698E70, &CCoverPoint::CanAccomodateAnotherPed);
+    ReversibleHooks::Install("CCoverPoint", "ReserveCoverPointForPed", 0x698EB0, &CCoverPoint::ReserveCoverPointForPed);
+    ReversibleHooks::Install("CCoverPoint", "ReleaseCoverPointForPed", 0x698EF0, &CCoverPoint::ReleaseCoverPointForPed);
+}
+
+// 0x698E70
+bool CCoverPoint::CanAccomodateAnotherPed() {
+    if (!m_nMaxPedsInCover)
+        return false;
+
+    if (m_nMaxPedsInCover != 2)
+        return !m_apCoveredPeds[0];
+
+    return !m_apCoveredPeds[0] || m_apCoveredPeds[1];
+}
+
+// 0x698EB0
+void CCoverPoint::ReserveCoverPointForPed(CPed* ped) {
+    for (int i = 0; i < MAX_COVER_PEDS; i++) {
+        if (CPed*& it = m_apCoveredPeds[i]) {
+            it = ped;
+            ped->RegisterReference(reinterpret_cast<CEntity**>(&it));
+            break;
+        }
+    }
+}
+
+// 0x698EF0
+void CCoverPoint::ReleaseCoverPointForPed(CPed* toRelease) {
+    if (!toRelease)
+        return; // NOTSA
```

Comment:
```suggestion
    if (!toRelease)
        return; // NOTSA
```
Not a good idea. Unless your code fixes a confirmed major crash/bug, there's no need to change the logic.

Reply:
Just look at the code, and think how this could easily crash it.
Perhaps I should change it to `assert` instead.

Reply:
I did check the code, and it won't crash. You can remove the assert.

Reply:
It will crash either way tbh, so yeah
Doesn't really matter